app/mailers/application_mailer.rb

```diff
@@ -15,7 +15,16 @@ class ApplicationMailer < Mail::Notify::Mailer
   end
 
   def notify_email(headers)
-    headers = headers.merge(rails_mailer: mailer_name, rails_mail_template: action_name)
+    set_mailer_details
     view_mail(GENERIC_NOTIFY_TEMPLATE, headers)
   end
+
+private
+
+  def set_mailer_details
+    message.instance_variable_set(:@rails_mailer, mailer_name)
+    message.instance_variable_set(:@rails_mail_template, action_name)
+    message.class.send(:attr_reader, :rails_mailer)
+    message.class.send(:attr_reader, :rails_mail_template)
```

Comment:
I don't love this but if this is what we have to do 🤷‍♂️

Reply:
The interceptor is where we need the data and it doesn't have access to `mailer_name` and `action_name`. You can get to `mailer_name` through `deployment_method` which is on the mailer class, but not `action_name`. I thought better to deal with them in the same way for clarity rather than get one from digging into `deployment_method` and the other set as an instance variable. 

Yes, lots of rabbit warrens.